tikv/raftstore/coprocessor.go

```diff
@@ -334,14 +349,57 @@ func newTableSplitChecker(firstEncodedTablePrefix, splitKey []byte, policy pdpb.
 	}
 }
 
+func isTableKey(encodedKey []byte) bool {
+	if tikv.IsShardingEnabled() {
+		return len(encodedKey) >= tablecodec.TableSplitKeyLen &&
+			tablecodec.TablePrefix()[0] == encodedKey[0]
+	}
+	return len(encodedKey) >= otablecodec.TableSplitKeyLen &&
+		otablecodec.TablePrefix()[0] == encodedKey[0]
+}
+
+func isSameTable(leftKey, rightKey []byte) bool {
+	if tikv.IsShardingEnabled() {
+		return isTableKey(leftKey) && isTableKey(rightKey) && bytes.Compare(leftKey[:tablecodec.TableSplitKeyLen],
+			rightKey[:tablecodec.TableSplitKeyLen]) == 0
+	}
+	return isTableKey(leftKey) && isTableKey(rightKey) && bytes.Compare(leftKey[:otablecodec.TableSplitKeyLen],
+		rightKey[:otablecodec.TableSplitKeyLen]) == 0
+}
+
+func extractTablePrefix(key []byte) []byte {
+	if tikv.IsShardingEnabled() {
+		return key[:tablecodec.TableSplitKeyLen]
+	} else {
+		return key[:otablecodec.TableSplitKeyLen]
+	}
+}
+
+// Feed keys in order to find the split key.
+// If currentEncodedKey doesn't belong to checker.firstEncodedTablePrefix, returns the encoded table prefix
+// of currentEncodedKey
 func (checker *tableSplitChecker) onKv(obCtx *observerContext, spCheckKeyEntry splitCheckKeyEntry) bool {
-	//Todo, currently it is a place holder
+	if len(checker.splitKey) != 0 {
+		return true
+	}
+	currentEncodedKey := OriginKey(spCheckKeyEntry.key)
+	var splitKey []byte
+	// Different tables or meet the very first table key of this region.
+	if !isSameTable(checker.firstEncodedTablePrefix, currentEncodedKey) || isTableKey(currentEncodedKey) {
```

Comment:
The logic is different than tikv.
Only when `firstEncodedTablePrefix` is empty, `isTableKey` sets `splitKey`